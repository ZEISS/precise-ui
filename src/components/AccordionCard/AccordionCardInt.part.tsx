import * as React from 'react';
import styled, { themed, css } from '../../utils/styled';
import { distance } from '../../distance';
import { StandardProps } from '../../common';
import { setLabels, getPropLabel, AccordionCardLabels } from '../../utils/labels';
import { getFontStyle } from '../../textStyles';

setLabels({
  openLabel: 'Open Details',
  closeLabel: 'Close Details',
});

const animationDuration = '0.3s';
const animationFunction = 'cubic-bezier(0, 0, 0.25, 1)';

export interface ActiveProps {
  open: boolean;
  onClick?(e: React.MouseEvent): void;
}

const StyledContainer = styled('div')<ActiveProps>(
  themed(
    ({
      open,
      theme: {
        accordionCard: { borderColor, openedBorderColor },
        text1,
      },
    }) => `
    ${getFontStyle({ size: 'medium' })}
    
    position: relative;
    color: ${text1};
    border: 1px solid ${open ? openedBorderColor : borderColor};
  `,
  ),
);

const StyledHeaderContainer = styled('div')<ActiveProps>(
  themed(
    ({
      open,
      theme: {
        accordionCard: { headerPadding, openedHeaderBorderColor, openedHeaderBackground, headerBackground },
      },
    }) => `
      padding: ${headerPadding};
      background: ${open ? openedHeaderBackground : headerBackground};
      border-bottom: ${open ? `1px solid ${openedHeaderBorderColor}` : 'none'};
    `,
  ),
);

export const StyledActionContainer = styled.div<ActiveProps>(
  themed(
    ({ open, theme: { ui0, ui1 } }) => css`
      text-align: center;
      padding: ${distance.small} ${distance.medium};
      cursor: pointer;
      background: ${open ? ui1 : ui0};
      border: ${open ? `1px solid ${ui0}` : 'none'};
      margin: -1px -1px -1px -1px;
      color: ${open ? ui0 : ui1};
    `,
  ),
);

const StyledDetailsContainerAnimator = styled('div')<ActiveProps>`
  transition: all ${animationDuration} ${animationFunction};
  max-height: ${({ open }) => (open ? '10000px' : '0')};
  opacity: ${({ open }) => (open ? '1' : '0')};
  overflow: hidden;
`;

export interface RenderActionEvent {
  opened: boolean;
}

export interface ActionClickEvent {
  opened: boolean;
}

export interface AccordionCardProps extends StandardProps, AccordionCardLabels {
  /**
   * Displays card header.
   */
  header: React.ReactNode;
  /**
   * Displayed content when card is opened.
   */
  children: React.ReactNode;
  /**
   * Determines if details are open for the card.
   */
  opened?: boolean;
  /**
   * Custom render for card action button.
   */
  renderAction?(e: RenderActionEvent): React.ReactNode;
  /**
   * Event fired when action button clicked.
   */
  onActionClick?(e: ActionClickEvent): void;
}

export interface AccordionCardState {
  opened: boolean;
  controlled: boolean;
}

export class AccordionCard extends React.Component<AccordionCardProps, AccordionCardState> {
  constructor(props: AccordionCardProps) {
    super(props);
    this.state = {
      opened: props.opened !== undefined ? props.opened : false,
      controlled: props.opened !== undefined,
    };
  }

  static getDerivedStateFromProps(props: AccordionCardProps, state: AccordionCardState) {
    return state.controlled ? { opened: props.opened } : {};
  }

  private handleClick = () => {
    const { onActionClick } = this.props;
    const { opened, controlled } = this.state;
    !controlled && this.setState({ opened: !opened });

    if (typeof onActionClick === 'function') {
      onActionClick({ opened: controlled ? opened : !opened });
    }
  };

  private renderDefaultAction = (_: RenderActionEvent) => {
    const { theme } = this.props;
    const { opened } = this.state;

    return (
      <StyledActionContainer theme={theme} onClick={this.handleClick} open={opened}>
        {getPropLabel(this.props, opened ? 'closeLabel' : 'openLabel')}
      </StyledActionContainer>
    );
  };

  private renderActions() {
    const { renderAction = this.renderDefaultAction } = this.props;
    const { opened } = this.state;
    return renderAction({ opened });
  }

  render() {
    const { onActionClick: _0, theme, children, header, ...props } = this.props;
    const { opened } = this.state;

    return (
      <StyledContainer {...props} theme={theme} open={opened}>
        <StyledHeaderContainer theme={theme} open={opened}>
          {header}
        </StyledHeaderContainer>
        <StyledDetailsContainerAnimator theme={theme} open={opened}>
          {children}
        </StyledDetailsContainerAnimator>
        {this.renderActions()}
      </StyledContainer>
    );
  }
}
