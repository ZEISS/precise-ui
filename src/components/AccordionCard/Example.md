Using the `AccordionCard` component. Renders card which is expandable by clicking action button.

```jsx
const { AccordionCard } = require('precise-ui');

<AccordionCard header="Title">
  <div style={{ padding: '10px' }}>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's
    standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a
    type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting,
    remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing
    Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of
    Lorem Ipsum.
  </div>
</AccordionCard>;
```

`AccordionCard` in controlled mode with custom action renderer.

```jsx
const { AccordionCard, StackPanel, StackItem, ActionLink } = require('precise-ui');

class AccordionCardInt extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      opened: false,
    };
  }

  renderHeader() {
    return (
      <StackPanel>
        <StackItem>Some Title</StackItem>
        <StackItem style={{ textAlign: 'right' }}>
          <ActionLink onClick={() => this.setState({ opened: !this.state.opened })}>
            {this.state.opened ? 'close' : 'open'}
          </ActionLink>
        </StackItem>
      </StackPanel>
    );
  }
  render() {
    return (
      <AccordionCard header={this.renderHeader()} renderAction={() => null} opened={this.state.opened}>
        <div style={{ padding: '10px' }}>
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's
          standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to
          make a type specimen book. It has survived not only five centuries, but also the leap into electronic
          typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset
          sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus
          PageMaker including versions of Lorem Ipsum.
        </div>
      </AccordionCard>
    );
  }
}

<AccordionCardInt />;
```
