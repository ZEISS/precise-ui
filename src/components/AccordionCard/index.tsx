export { AccordionCard, AccordionCardProps } from './AccordionCardInt.part';
