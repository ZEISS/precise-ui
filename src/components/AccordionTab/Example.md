**Elementary**

Using the accordion tabs component with a single tab given in form of a `AccordionTab` element. The content is directly rendered, i.e., we could also have some more complex structure inside.

```jsx
import { Accordion, AccordionTab } from 'precise-ui';

<Accordion>
  <AccordionTab header="Title of the tab">
    <em>Content of the tab</em>
  </AccordionTab>
</Accordion>
```

The accordion tab header can also be a complex element. It is possible to pass in a `ReactChild` as header in order to provide custom header.

```jsx
import { Accordion, AccordionTab, colors } from 'precise-ui';

<Accordion>
  <AccordionTab header={<span style={{ color: colors.blue }}>Hello</span>}>
    Content of the tab
  </AccordionTab>
</Accordion>
```
