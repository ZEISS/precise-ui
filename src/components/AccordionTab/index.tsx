import * as React from 'react';
import { StandardProps } from '../../common';

export interface AccordionTabProps extends StandardProps {
  /**
   * The title of the tab in the accordion selection.
   */
  header: React.ReactChild;
  /**
   * The content of the tab to be rendered if open.
   */
  children?: React.ReactNode;
}

/**
 * The accordion tab is used to define a single tab within a accordion component.
 */
export const AccordionTab: React.SFC<AccordionTabProps> = ({ header: _0, ...props }) => <div {...props} />;
AccordionTab.displayName = 'AccordionTab';
