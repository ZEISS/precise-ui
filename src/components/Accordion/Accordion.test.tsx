import * as React from 'react';
import * as enzyme from 'enzyme';
import { Accordion } from './';
import { AccordionBasic } from './AccordionBasic.part';
import { AccordionCard } from './AccordionCard.part';
import { AccordionTab } from '../AccordionTab';

describe('<Accordion />', () => {
  it('should render an empty <Accordion> component', () => {
    const wrapper = enzyme.shallow(<Accordion />);
    const nodes = wrapper.filter((node: any) => node.text() !== '');
    expect(nodes.exists()).toBeFalsy();
  });

  it('should render <AccordionBasic> with accordion mode', () => {
    const wrapper = enzyme.mount(
      <Accordion mode="accordion">
        <AccordionTab header="Header">Content</AccordionTab>
      </Accordion>,
    );
    expect(wrapper.find(AccordionBasic).length > 0).toBeTruthy();
  });

  it('should render <AccordionCard> with card mode', () => {
    const wrapper = enzyme.mount(
      <Accordion mode="card">
        <AccordionTab header="Header">Content</AccordionTab>
      </Accordion>,
    );
    expect(wrapper.find(AccordionCard).length > 0).toBeTruthy();
  });

  it('should render a <Accordion> component with several tabs', () => {
    const wrapper = enzyme.mount(
      <Accordion>
        <AccordionTab header="First">First content.</AccordionTab>
        <AccordionTab header="Second">Second content.</AccordionTab>
        <AccordionTab header="Third">Third content.</AccordionTab>
        <AccordionTab header="Last">Final content.</AccordionTab>
      </Accordion>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Accordion> component with several tabs in and selectedIndex', () => {
    const wrapper = enzyme.mount(
      <Accordion selectedIndex={1}>
        <AccordionTab header="First">First content.</AccordionTab>
        <AccordionTab header="Second">Second content.</AccordionTab>
        <AccordionTab header="Last">Final content.</AccordionTab>
      </Accordion>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
