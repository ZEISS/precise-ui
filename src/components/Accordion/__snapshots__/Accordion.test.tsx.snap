// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render a <Accordion> component with several tabs 1`] = `
.c5 {
  display: inline-block;
}

.c5 > svg {
  float: left;
}

.c0 {
  padding: 0;
  list-style: none;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  position: relative;
  z-index: 1;
  margin: 0;
  text-align: left;
  overflow: visible;
  border-top: 1px solid rgba(211,211,211,1);
  box-sizing: border-box;
  color: rgba(62,62,62,1);
}

.c1:last-child {
  border-bottom: 1px solid rgba(211,211,211,1);
}

.c2 {
  -webkit-transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  background-color: rgba(255,255,255,1);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  white-space: pre-wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1rem;
}

.c4 {
  font-size: 1.125rem;
  margin-right: 1rem;
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<Responsive>
  <ModeProvider
    dimensions={
      Object {
        "angle": 0,
        "height": 768,
        "width": 1024,
      }
    }
    innerRef={[Function]}
  >
    <AccordionInt
      dimensions={
        Object {
          "angle": 0,
          "height": 768,
          "width": 1024,
        }
      }
      innerRef={[Function]}
      mode="accordion"
    >
      <div>
        <AccordionBasic
          dimensions={
            Object {
              "angle": 0,
              "height": 768,
              "width": 1024,
            }
          }
        >
          <styled.div>
            <div
              className=""
            >
              <styled.ul>
                <ul
                  className="c0"
                >
                  <styled.li
                    key="item-0"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              First
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                  <styled.li
                    key="item-1"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              Second
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                  <styled.li
                    key="item-2"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              Third
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                  <styled.li
                    key="item-3"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              Last
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
            </div>
          </styled.div>
        </AccordionBasic>
      </div>
    </AccordionInt>
  </ModeProvider>
</Responsive>
`;

exports[`<Accordion /> should render a <Accordion> component with several tabs in and selectedIndex 1`] = `
.c5 {
  display: inline-block;
}

.c5 > svg {
  float: left;
}

.c8 {
  overflow: hidden;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  height: 0;
}

.c0 {
  padding: 0;
  list-style: none;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  position: relative;
  z-index: 1;
  margin: 0;
  text-align: left;
  overflow: visible;
  border-top: 1px solid rgba(211,211,211,1);
  box-sizing: border-box;
  color: rgba(62,62,62,1);
}

.c1:last-child {
  border-bottom: 1px solid rgba(211,211,211,1);
}

.c2 {
  -webkit-transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  background-color: rgba(255,255,255,1);
}

.c6 {
  -webkit-transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  transition: background-color 0.3s cubic-bezier(0,0,0.25,1);
  background-color: rgba(248,248,248,1);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  white-space: pre-wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1rem;
}

.c4 {
  font-size: 1.125rem;
  margin-right: 1rem;
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c7 {
  font-size: 1.125rem;
  margin-right: 1rem;
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  transition: transform 0.2s cubic-bezier(0,0,0.25,1);
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c9 {
  padding: 0.5rem 1rem 2rem 3.125rem;
}

<Responsive
  selectedIndex={1}
>
  <ModeProvider
    dimensions={
      Object {
        "angle": 0,
        "height": 768,
        "width": 1024,
      }
    }
    innerRef={[Function]}
    selectedIndex={1}
  >
    <AccordionInt
      dimensions={
        Object {
          "angle": 0,
          "height": 768,
          "width": 1024,
        }
      }
      innerRef={[Function]}
      mode="accordion"
      selectedIndex={1}
    >
      <div>
        <AccordionBasic
          dimensions={
            Object {
              "angle": 0,
              "height": 768,
              "width": 1024,
            }
          }
          selectedIndex={1}
        >
          <styled.div>
            <div
              className=""
            >
              <styled.ul>
                <ul
                  className="c0"
                >
                  <styled.li
                    key="item-0"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              First
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                  <styled.li
                    key="item-1"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={true}
                      >
                        <div
                          className="c6"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={true}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c7"
                                  isRotated={true}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c7"
                                    isRotated={true}
                                  >
                                    <i
                                      className="c5 c7"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              Second
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={true}
                        timeout={300}
                        unmountOnExit={true}
                      >
                        <styled.div
                          expandedHeight={0}
                          status="entering"
                          style={
                            Object {
                              "minHeight": undefined,
                            }
                          }
                          timeout={300}
                        >
                          <div
                            className="c8"
                            style={
                              Object {
                                "minHeight": undefined,
                              }
                            }
                          >
                            <div>
                              <styled.div>
                                <div
                                  className="c9"
                                >
                                  <AccordionTab
                                    header="Second"
                                  >
                                    <div>
                                      Second content.
                                    </div>
                                  </AccordionTab>
                                </div>
                              </styled.div>
                            </div>
                          </div>
                        </styled.div>
                      </Expander>
                    </li>
                  </styled.li>
                  <styled.li
                    key="item-2"
                  >
                    <li
                      className="c1"
                    >
                      <styled.div
                        active={false}
                      >
                        <div
                          className="c2"
                        >
                          <styled.div
                            onClick={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                            >
                              <Styled(Icon)
                                isRotated={false}
                                name="KeyboardArrowRight"
                              >
                                <Icon
                                  className="c4"
                                  isRotated={false}
                                  name="KeyboardArrowRight"
                                >
                                  <styled.i
                                    className="c4"
                                    isRotated={false}
                                  >
                                    <i
                                      className="c5 c4"
                                    >
                                      <KeyboardArrowRight
                                        height="1em"
                                        style={
                                          Object {
                                            "fill": "currentColor",
                                            "stroke": undefined,
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                        width="1em"
                                      >
                                        <svg
                                          height="1em"
                                          style={
                                            Object {
                                              "fill": "currentColor",
                                              "stroke": undefined,
                                            }
                                          }
                                          viewBox="0 0 24 24"
                                          width="1em"
                                        >
                                          <path
                                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                                          />
                                        </svg>
                                      </KeyboardArrowRight>
                                    </i>
                                  </styled.i>
                                </Icon>
                              </Styled(Icon)>
                              Last
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <Expander
                        expand={false}
                        timeout={300}
                        unmountOnExit={true}
                      />
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
            </div>
          </styled.div>
        </AccordionBasic>
      </div>
    </AccordionInt>
  </ModeProvider>
</Responsive>
`;
